### PR TITLE
docs: sync documentation to latest changes

### PR DESCRIPTION
**Pull request by fncolon**
### Pull Request Checklist

- [ ] Have you added new tests to prevent regressions?
- [ ] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [ ] Did you update the typescript typings accordingly (if applicable)?
- [ ] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [ ] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)?

### Description Of Change

- Since we have version picker, we don't need documentation section anymore, and some additional changes to keep README.md simpler as much as possible.

- Adding repository discussions as Sequelize resource

- Remove dollar sign so copy is much easy

### CODE REVIEW

**Comment by ephys** [comment]
I agree with most changes but you should keep the documentation section even if it's just a small paragraph redirecting to https://sequelize.org/

**Comment by fncolon** [comment]
> I agree with most changes but you should keep the documentation section even if it's just a small paragraph redirecting to https://sequelize.org/

It's avoid future changes for chaining version inside documentation section and different latest version for https://sequelize.org/master

How about

```md
## Documentation

- [Latest Version Documentation](https://sequelize.org/master)
- [Homepage](https://sequelize.org)
- [Contributing](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)
```

**Comment by ephys** [comment]
I think we use `/main` instead of `/master` for the documentation too now.

If we include the latest documentation version, we should specify for which version it is for (v7) to reduce risk of confusion.

And I propose "All documentation versions" instead of "Homepage" so users not looking for v7 have a easier time finding it :)

**Comment by fncolon** [comment]
> I think we use `/main` instead of `/master` for the documentation too now.
> 
> If we include the latest documentation version, we should specify for which version it is for (v7) to reduce risk of confusion.
> 
> And I propose "All documentation versions" instead of "Homepage" so users not looking for v7 have a easier time finding it :)

Noted

**Comment by fncolon** [comment]
Another solution to reduce risk of confusion, maybe to provide `/latest` which is will automatic redirecting to latest version of Sequelize.

File: README.md [hunk 1]
```diff
@@ -33,33 +33,28 @@ We have recently created an [OpenCollective based money pool](https://opencollec
...
 
 If you have security issues to report, please refer to our [Responsible Disclosure Policy](https://github.com/sequelize/sequelize/blob/main/SECURITY.md) for more details.
 
 ## Resources
 
 - [Changelog](https://github.com/sequelize/sequelize/releases)
+- [Discussions](https://github.com/sequelize/sequelize/discussions)
 - [Slack Inviter](http://sequelize-slack.herokuapp.com/)
-- [Stack Overflow](https://stackoverflow.com/questions/tagged/sequelize.js)
+- [StackOverflow](https://stackoverflow.com/questions/tagged/sequelize.js)
```
Comment (WikiRik) [inline README.md:57]:
```suggestion
- [Stack Overflow](https://stackoverflow.com/questions/tagged/sequelize.js)
```

Stack Overflow uses its name with a space

**Comment by ephys** [comment]
> If we include the latest documentation version, we should specify for which version it is for (v7) to reduce risk of confusion.

Sorry, what I meant was add "v7" to the link name, especially since it's a pre-release and everyone will stay on v6 until it's ready :)

**Comment by fncolon** [comment]
> > If we include the latest documentation version, we should specify for which version it is for (v7) to reduce risk of confusion.
> 
> Sorry, what I meant was add "v7" to the link name, especially since it's a pre-release and everyone will stay on v6 until it's ready :)

can you elaborate about link name?

**Comment by ephys** [comment]
Simply "Latest Version Documentation (v7)" instead of "Latest Version Documentation"

**Comment by sdepold** [comment]
Why do we remove the changelog and and links? I think they are really useful

**Comment by fncolon** [comment]
> Why do we remove the changelog and and links? I think they are really useful

Sorry, but I didn't remove the changelog.
Alreayd revert the links too.

**Pull request merged**